### PR TITLE
VIH-8213 added logic to handle participant transfer from room type

### PR DESCRIPTION
**Pull request by omio-ma**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/VIH-8213

### Change description ###
- updated logic to handle quick link user being in consultation room when being transferred to hearing

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: VideoApi/VideoApi.UnitTests/Controllers/ConferenceManagement/TransferParticipantTests.cs [hunk 1]
```diff
@@ -128,5 +128,30 @@ public async Task should_return_kinly_status_code_on_error()
             typedResult.StatusCode.Should().Be(statusCode);
             typedResult.Value.Should().Be(response);
         }
+
+        [Test]
+        [TestCase(UserRole.QuickLinkObserver)]
+        [TestCase(UserRole.QuickLinkParticipant)]
+        public async Task Room_To_Transfer_From_Is_Not_Set_To_Waiting_Room_When_Participant_CurrentConsultationRoom_Is_Defined(UserRole userRole)
```
Comment (FarronF) [inline VideoApi/VideoApi.UnitTests/Controllers/ConferenceManagement/TransferParticipantTests.cs:135]: Make the name about what we expect not what we don't expect.

File: VideoApi/VideoApi/Controllers/ConferenceManagementController.cs [hunk 2]
```diff
@@ -187,5 +187,10 @@ public async Task<IActionResult> TransferParticipantAsync(Guid conferenceId, Tra
                 return StatusCode(ex.StatusCode, ex.Response);
             }
         }
+
+        private string TransferFromRoomType(ParticipantBase participant)
+        {
+            return (participant.UserRole == UserRole.QuickLinkParticipant || participant.UserRole == UserRole.QuickLinkObserver) && participant.CurrentConsultationRoom != null ? participant.CurrentConsultationRoom.Label : RoomType.WaitingRoom.ToString();
```
Comment (FarronF) [inline VideoApi/VideoApi/Controllers/ConferenceManagementController.cs:193]: I wonder if we could just have the second part of this check? I get that we only care about quick link users in this ticket.

I'll leave that up to you. If we keep this we should test when it is a quick link user and the consultation room is null that we get the waiting room string

File: VideoApi/VideoApi/Controllers/ConferenceManagementController.cs [hunk 3]
```diff
@@ -190,7 +190,7 @@ public async Task<IActionResult> TransferParticipantAsync(Guid conferenceId, Tra
 
         private string TransferFromRoomType(ParticipantBase participant)
         {
-            return (participant.UserRole == UserRole.QuickLinkParticipant || participant.UserRole == UserRole.QuickLinkObserver) && participant.CurrentConsultationRoom != null ? participant.CurrentConsultationRoom.Label : RoomType.WaitingRoom.ToString();
+            return (participant.UserRole == UserRole.QuickLinkParticipant || participant.UserRole == UserRole.QuickLinkObserver) && participant.CurrentConsultationRoom != null && !string.IsNullOrWhiteSpace(participant.CurrentConsultationRoom.Label) ? participant.CurrentConsultationRoom.Label : RoomType.WaitingRoom.ToString();
```
Comment (FarronF) [inline VideoApi/VideoApi/Controllers/ConferenceManagementController.cs:193]:
```suggestion
            return (participant.UserRole == UserRole.QuickLinkParticipant || participant.UserRole == UserRole.QuickLinkObserver) && !string.IsNullOrWhiteSpace(participant.CurrentConsultationRoom?.Label) ? participant.CurrentConsultationRoom.Label : RoomType.WaitingRoom.ToString();
```

Comment (FarronF) [reply VideoApi/VideoApi/Controllers/ConferenceManagementController.cs:193]: Would this work?

Comment (omio-ma) [reply VideoApi/VideoApi/Controllers/ConferenceManagementController.cs:193]: I can try this, but I felt the other way to write down this logic was a bit more readable

Comment (omio-ma) [reply VideoApi/VideoApi/Controllers/ConferenceManagementController.cs]: all the unit test passed, one of the test was checking when the currentconsutationroom property is null so I guess it should work

**Review by FarronF — APPROVED** [review]

**Review by krishnapottekkat-hmcts — APPROVED** [review]

**Pull request merged**